### PR TITLE
Update blockio-uring

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

**Review by jorisdral — APPROVED** [review]

File: fs-api-blockio/test/Main.hs [hunk 1]
```diff
@@ -44,12 +42,16 @@ instance Arbitrary ByteString where
   arbitrary = BS.pack <$> arbitrary
   shrink = fmap BS.pack . shrink .  BS.unpack
 
-fromByteString :: PrimMonad m => ByteString -> m (MutableByteArray (PrimState m))
-fromByteString bs = thawByteArray (ByteArray ba) 0 (SBS.length sbs)
-  where !sbs@(SBS.SBS ba) = SBS.toShort bs
+fromByteStringPinned :: PrimMonad m => ByteString -> m (MutableByteArray (PrimState m))
+fromByteStringPinned bs = do
+  mba <- newPinnedByteArray (BS.length bs)
+  forM_ (zip [0..] (BS.unpack bs)) $ \(i, x) -> writeByteArray mba i x
```
Comment (dcoutts) [inline fs-api-blockio/test/Main.hs:48]: this could be faster, but perhaps it doesn't matter

File: lsm-tree.cabal [hunk 2]
```diff
@@ -420,6 +420,18 @@ test-suite lsm-prototypes-tests
     , lsm-tree:prototypes
     , tasty
 
+-- TODO: temporary flag that overrides the operating system dependents
```
Comment (dcoutts) [inline lsm-tree.cabal:423]: The flag doesn't have to be temporary. It's just the use of the flag in the project file that should be temporary.

File: lsm-tree.cabal [hunk 3]
```diff
@@ -420,6 +420,18 @@ test-suite lsm-prototypes-tests
...
+-- implementation of the HasBlockIO interface, using the serial implementation
+-- intead. This allows for fixing bugs in the blockio-uring implementation
+-- without blocking parallel development efforts that rely on HasBlockIO but not
+-- on its exact implementation. One could argue for keeping this flag even after
+-- blockio-uring has become stable enough, but I (Joris) would prefer to not
+-- expose cabal flags to the user.
```
Comment (dcoutts) [inline lsm-tree.cabal:429]: Ok ok, it's arguable.

**Review by dcoutts — APPROVED** [review]
LGTM

File: fs-api-blockio/test/Main.hs (hunk 1 above)
Comment (jorisdral) [reply fs-api-blockio/test/Main.hs:48]: It's slow but straightforward. The bytestrings the tests are generating are relatively small

File: cabal.project [hunk 4]
```diff
@@ -39,11 +39,15 @@ package lsm-tree
   -- relevant mostly only for development & testing
   ghc-options: -fno-ignore-asserts
 
+  -- TODO: temporarily disable asynchronous HasBlockIO on Linux-based systems.
+  -- See lsm-tree#129.
+  flags: +serialblockio
```
Comment (phadej) [inline cabal.project:44]: this flag doesn't need to be enabled for all, and especially not for CI.

Currently it's hiding some bug somewhere.

Comment (jorisdral) [reply cabal.project:44]: I'm looking into the bug, but I didn't want to have that hold back other PRs

**Pull request merged**